src/services/CertificateGenerationService.ts

```diff
@@ -889,36 +884,24 @@ class CertificateGenerationService {
       }),
     };
     let defects: IDefectParent[] = [];
-    return this.lambdaClient
-      .invoke(invokeParams)
-      .then(
-        (
-          response: PromiseResult<Lambda.Types.InvocationResponse, AWSError>
-        ) => {
-          const payload: any =
-            this.lambdaClient.validateInvocationResponse(response);
-          defects = JSON.parse(payload.body);
-
-          if (!defects || defects.length === 0) {
-            throw new HTTPError(
-              400,
-              `${ERRORS.LAMBDA_INVOCATION_BAD_DATA} ${JSON.stringify(payload)}.`
-            );
-          }
-
-          console.log(
-            `Successfully retrieved ${defects.length} welsh defects translations.`
-          );
+    let retries = 0;
+    while (retries < 3) {
+        try {
+            const response: PromiseResult<Lambda.Types.InvocationResponse, AWSError> = await this.lambdaClient.invoke(invokeParams);
+            const payload: any = this.lambdaClient.validateInvocationResponse(response);
+            const defectsParsed = JSON.parse(payload.body);
 
-          return defects;
+            if (!defectsParsed || defectsParsed.length === 0) {
+                throw new HTTPError(400, `${ERRORS.LAMBDA_INVOCATION_BAD_DATA} ${JSON.stringify(payload)}.`);
+            }
+            defects = defectsParsed;
+            return defects;
+        } catch (error) {
+            retries++;
+            console.error(`There was an error retrieving the welsh defect translations: ${error}`);
```

Comment:
Can you include the attempt number on the error log, will help with support and to identify if the retry is working as expected